xmppserver/src/main/java/org/jivesoftware/openfire/net/SocketReader.java

```diff
@@ -370,7 +370,7 @@ private IQ getIQ(Element doc) {
             return new Roster(doc);
         }else if (query != null && "jabber:iq:version".equals(query.getNamespaceURI())) {
             IQ iq = new IQ(doc);
-            if (iq.getType().equals(IQ.Type.result) && !iq.getFrom().toString().contains("@") && !(StringUtils.countMatches(iq.getFrom().toString(),".")>1)){
+            if (iq.getType().equals(IQ.Type.result) && !iq.getFrom().toString().contains("@") && iq.getFrom().equals(session.getAddress())){
```

Comment:
@guusdk I think the JID of the remote server. Maybe it's already available in the connection field for the current session connection. I can access it with` session.getAddress()`

Reply:
Did you test if it does? If so, then excellent!